experimental/Schemes/IdealSheaves.jl

```diff
@@ -293,36 +293,34 @@ end
 @doc raw"""
     simplify!(I::AbsIdealSheaf)
 
-Replaces the set of generators of the ideal sheaf by a minimal
-set of random linear combinations in every affine patch.
+Replace the ideals on the `affine_patches` of the `scheme` of `I` 
+with isomorphic ideals generated by a small set of generators.
 """
-function simplify!(I::AbsIdealSheaf)
-  new_ideal_dict = IdDict{AbsAffineScheme, Ideal}()
-  for U in basic_patches(default_covering(space(I)))
-    new_ideal_dict[U] = ideal(OO(U), small_generating_set(I(U)))
-    #=
-    n = ngens(I(U))
-    n == 0 && continue
-    R = ambient_coordinate_ring(U)
-    kk = coefficient_ring(R)
-    new_gens = elem_type(OO(U))[]
-    K = ideal(OO(U), new_gens)
-    while !issubset(I(U), K)
-      new_gen = dot([rand(kk, 1:100) for i in 1:n], gens(I(U)))
-      while new_gen in K
-        new_gen = dot([rand(kk, 1:100) for i in 1:n], gens(I(U)))
-      end
-      push!(new_gens, new_gen)
-      K = ideal(OO(U), new_gens)
-    end
-    Oscar.object_cache(underlying_presheaf(I))[U] = K
-    =#
+function simplify!(I::IdealSheaf, cov::Covering=default_covering(space(I)))
+  object_cache = I.I.obj_cache
+
+  for U in basic_patches(cov)
+    #if !any(U===i for i in keys(object_cache))
+    #  continue
+    #end
+    object_cache[U] = ideal(OO(U), small_generating_set(I(U)))
   end
-  I.I.obj_cache = new_ideal_dict # for some reason the line below led to compiler errors.
-  #Oscar.object_cache(underlying_presheaf(I)) = new_ideal_dict
   return I
 end
 
+@doc raw"""
+    simplify(I::IdealSheaf)
+
+Replace the generators of the ideals on the `affine_patches` of the `scheme` of `I` 
+by a small set of generators.
+"""
+function simplify(I::IdealSheaf, cov::Covering=default_covering(scheme(I)))
+  id_dict = IdDict{AbsAffineScheme, Ideal}(
+                U => ideal(OO(U), small_generating_set(saturated_ideal(I(U)))) for U in patches(cov)
```

Comment:
Why is there a saturated ideal here?
Shouldn't this be taken care off by the `small_generating_set` function instead?

Reply:
Currently `small_generating_set` does not have a method for this signature. But we could introduce one for localizations at D(f), which uses saturated_ideal internally.

Reply:
`small_generating_set` does not compute a saturation anymore. It's not what the name of the function suggests and it killed performance in some cases. So I threw it away.